Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3709,20 +2785,34 @@ - (void)yapDatabaseModified:(NSNotification *)notification
     __block BOOL scrollToBottom = wasAtBottom;
 
     [self.collectionView performBatchUpdates:^{
+        // TODO: We need to reload neighbords of changed cells.
```

Comment:
I'm not 100% of the best way to do this yet, so I'm deferring this work until another PR.

Reply:
We _did_ do it before, but we were doing it wrong.  

Consider this rule: `If you send two messages, only show recipient status for the second one.`

We used to ensure that behavior using the old `lastDeliveredMessage` property, which I've removed.  

* It was updating the first message when the second was sent by touching the "last sent message". 
* This generated extra "database modified" chatter.
* It also only handled the simplest case.  If you deleted the second message, the status would not appear on the first message.

Note that there's another rule that we also weren't enforcing correctly: `only show date/time header for the first message in a "block" of message.`

See my next PR for my first attempt to arrive at a more general, correct and maintainable solution.